android/src/main/java/cl/json/social/TargetChosenReceiver.java

```diff
@@ -0,0 +1,91 @@
+package cl.json.social;
+
+import android.annotation.TargetApi;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.IntentSender;
+import android.os.Build;
+
+import com.facebook.react.bridge.Callback;
+import com.facebook.react.bridge.ReactContext;
+
+/**
+ * Receiver to record the chosen component when sharing an Intent.
+ */
+public class TargetChosenReceiver extends BroadcastReceiver {
+    private static final String EXTRA_RECEIVER_TOKEN = "receiver_token";
+    private static final Object LOCK = new Object();
+
+    private static String sTargetChosenReceiveAction;
+    private static TargetChosenReceiver sLastRegisteredReceiver;
+
+    private static Callback successCallback;
+    private static Callback failureCallback;
+
+    static boolean isSupported() {
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1;
+    }
+
+    public static void registerCallbacks(Callback success, Callback failure) {
+        successCallback = success;
+        failureCallback = failure;
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP_MR1)
+    public static IntentSender getSharingSenderIntent(ReactContext reactContext) {
+        synchronized (LOCK) {
+            if (sTargetChosenReceiveAction == null) {
+                sTargetChosenReceiveAction = reactContext.getCurrentActivity().getPackageName() + "/" + TargetChosenReceiver.class.getName() + "_ACTION";
+            }
+            Context context = reactContext.getCurrentActivity().getApplicationContext();
```

Comment:
I have seen in other places that the current activity can sometimes go null on you in react-native, unexpectedly. Is there a reason not to use the ReactContext directly? https://github.com/facebook/react-native/issues/18345

Basically if you just need the context you shouldn't have to chain through the activity to get it if I recall correctly

Reply:
Yes, that's my bad. We only need context here. So, reactContext can be used directly. I will make the changes and make a commit regarding this.

Reply:
I have updated the code to use reactContext as it is.